### PR TITLE
kie-kogito-docs-631: Operator -> Cloud -> Deploy Supporting Services Section Updates

### LINKED ISSUE #631: Operator -> Cloud -> Deploy Supporting Services Section Updates
This section needs updates, I think it can be structured a bit better. I deserves better introduction, jobs service looks like un-linked while it's a supporting service too.

### PR DESCRIPTION
**Pull request by wmedvede**
Closes #631 

**Description:**

Please make sure that your PR meets the following requirements:

- [X] You have read the [contributions doc](https://github.com/apache/incubator-kie-kogito-docs/blob/main/CONTRIBUTING.md)
- [X] Pull Request title is properly formatted: `Issue-XYZ Subject`
- [X] Pull Request title contains the target branch if not targeting main: `[0.9.x] Issue-XYZ Subject`
- [X] The nav.adoc file has a link to this guide in the proper category
- [X] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 1]
```diff
@@ -6,210 +6,259 @@
...
-.Example of a SonataFlowPlatform instance with a Data Index deployment persisted to a postgresql database
-[source,yaml,subs="attributes+"]
+Additionally, using the `SonataFlowClusterPlatform` CR it's possible to configure a <<cluster-scoped-deployment, cluster scoped deployment>> of the supporting services.
+
+== Configuring the Supporting Services Persistence
+
+[#ephemeral-persistence-configuration]
+=== Ephemeral persistence configuration
+
+The ephemeral persistence of a service is supported by and embedded PostgreSQL database dedicated to it. That database, is re-created by the operator on every service restart.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:96]:
```suggestion
The ephemeral persistence of a service is supported by an embedded PostgreSQL database dedicated to it. That database, is re-created by the operator on every service restart.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 2]
```diff
@@ -6,210 +6,259 @@
...
-<3> PostgreSql JDBC URL
-<4> Number of Data Index pods, defaults to `1`
-<5> Custom Data Index container image name, if customization is required
 
+[#common-persistence-configuration]
+=== Common PostgreSQL persistence configuration
 
-[#deploy-job-service]
-=== Job Service
+To configure common a PostgreSQL service instance for all the supporting services you must read, xref:cloud/operator/using-persistence.adoc#configuring-persistence-using-the-sonataflowplatform-cr[Configuring the persistence using the SonataFlowPlatform CR].
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:168]:
```suggestion
To configure a common PostgreSQL service instance for all the supporting services you must read, xref:cloud/operator/using-persistence.adoc#configuring-persistence-using-the-sonataflowplatform-cr[Configuring the persistence using the SonataFlowPlatform CR].
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 3]
```diff
@@ -6,210 +6,259 @@
...
 
-You can deploy Job Service via `SonataFlowPlatform` configuration. The operator will then configure all new workflows, with the "preview" profile, to use that Job Service.
+In that case, the {operator_name} will automatically connect any of the supporting services with that common server configured in the field `spec.persistence`. And, similarly to the workflow's persistence, the following precedence rules apply:
 
-Following is a basic configuration. It will deploy an ephemeral Job Service to the same namespace as the `SonataFlowPlatform`.
+* If a supporting service, has a configured persistence, for example, the field `services.dataIndex.persistence` is configured. That configuration will apply.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:172]:
```suggestion
* If a supporting service has a configured persistence, for example, the field `services.dataIndex.persistence` is configured. That configuration will apply.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 4]
```diff
@@ -6,210 +6,259 @@
...
-----
-apiVersion: sonataflow.org/v1alpha08
-kind: SonataFlowPlatform
-metadata:
-  name: sonataflow-platform
-spec:
-  services:
-    jobService: {}
-----
+* If a supporting service, has no configured persistence, for example, the field `services.dataIndex.persistence` is not set at all, the persistence configuration will be taken from the current platform.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:174]:
```suggestion
* If a supporting service has no configured persistence, for example, the field `services.dataIndex.persistence` is not set at all, the persistence configuration will be taken from the current platform.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 5]
```diff
@@ -6,210 +6,259 @@
...
-<5> Custom Job Service container image name, if customization is required
+<1> Number of replicas. Defaults to 1. In the case of the jobService this value is always overridden to 1 by the operator, since that service is a singleton service.
+<2> Holds the particular configurations for the container that will execute the given supporting service.
+<3> Standard Kubernetes `env` configuration. This can be useful in cases where you need to fine tune any of the supporting services properties.
+<4> Standard Kubernetes `image` configuration. This can be useful in cases where you need to use an updated image for any of the supporting services.
+<5> Standard Kubernetes `initContainers` for the Pod that executes the supporting service.
+
+[NOTE]
+====
+The `podTemplate` field of any supporting has the majority of fields defined in the default Kubernetes PodSpec API.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:217]:
```suggestion
The `podTemplate` field of any supporting service has the majority of fields defined in the default Kubernetes PodSpec API.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 6]
```diff
@@ -6,210 +6,259 @@
...
+The same Kubernetes API validation rules applies to these fields.
```
Comment (nmirasch) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:218]:
```suggestion
The same Kubernetes API validation rules apply to these fields.
```

**Review by nmirasch — APPROVED** [review]
LGTM!

**Comment by wmedvede** [comment]
Hi @kaldesai , would you mind review?
Thanks!

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/install-serverless-operator.adoc [hunk 7]
```diff
@@ -140,6 +140,6 @@ The URL should be the same as the one you used when installing the operator.
 
 * xref:cloud/operator/known-issues.adoc[]
 * xref:cloud/operator/developing-workflows.adoc[]
-* xref:cloud/operator/enabling-jobs-service.adoc[]
+* xref:cloud/operator/supporting-services.adoc[Deploying the Supporting Services with the Operator]
```
Comment (ricardozanini) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/install-serverless-operator.adoc:143]:
```suggestion
* xref:cloud/operator/supporting-services.adoc[Deploying Supporting Services with the Operator]
```

**Review by ricardozanini — APPROVED** [review]

**Review by domhanak — APPROVED** [review]

**Pull request merged**